addons/website_event/models/event_event.py

```diff
@@ -376,8 +376,11 @@ def _create_menu(self, sequence, name, url, xml_id, menu_type):
             page_result = self.env['website'].sudo().new_page(
                 name=name + ' ' + self.name, template=xml_id,
                 add_menu=False, ispage=False)
-            url = "/event/" + slug(self) + "/page" + page_result['url']  # url contains starting "/"
-            view_id = page_result['view_id']
+            view_id = page_result["view_id"]
+            view = self.env["ir.ui.view"].browse(view_id)
+            url = (
+                "/event/" + slug(self) + "/page/" + view.key.split(".")[-1]
+            )  # url contains starting "/"
```

Comment:
Seems fine to inline, not sure I get the comment
```suggestion
            url = "event/" + slug(self) + "/page/" + view.key.split(".")[-1]
```